extensions/email/src/main/java/org/mule/extension/email/internal/EmailProperties.java

```diff
@@ -0,0 +1,189 @@
+
+    /**
+     * Creates a new instance and set all the properties required by the specified {@code protocol}.
+     */
+    private EmailProperties(String protocol, String host, String port, long connectionTimeout, long readTimeout, long writeTimeout)
+    {
+        this.properties = new Properties();
+        this.protocol = protocol;
+
+        set(PORT_PROPERTY, port);
+        set(HOST_PROPERTY, host);
+
+        if (isSecure())
+        {
+            set(START_TLS_PROPERTY, "true");
+            set(SOCKET_FACTORY_FALLBACK_PROPERTY, "false");
+            //set(SOCKET_FACTORY_CLASS_PROPERTY, "className");
+        }
+
+        set(CONNECTION_TIMEOUT_PROPERTY, Long.toString(connectionTimeout < 0L ? CONNECTION_TIMEOUT : readTimeout));
+        set(TIMEOUT_PROPERTY, Long.toString(readTimeout < 0L ? READ_TIMEOUT : readTimeout));
+
+        // Note: "mail." + protocol + ".writetimeout" breaks TLS/SSL Dummy Socket and makes tests run 6x slower!!!
+        if (writeTimeout > 0L)
+        {
+            set(WRITE_TIMEOUT_PROPERTY, Long.toString(writeTimeout));
+        }
+
+        set(TRANSPORT_PROTOCOL, protocol);
+    }
+
+
+    /**
+     * @return true if the protocol is secure, false otherwise.
+     */
+    private boolean isSecure()
+    {
+        return protocol.endsWith("s");
+    }
+
+    /**
+     * Sets a value to the specified property.
+     *
+     * @param property the property to be set.
+     * @param value    the corresponding value for the specified {@code property}
+     */
+    private void set(String property, String value)
+    {
+        properties.setProperty(String.format(property, protocol), value);
+    }
+
+    /**
+     * Configures a new properties instance with the specified parameters.
+     *
+     * @param protocol   the default protocol to be used.
+     * @param host       the host of the mail server.
+     * @param port       the port of the mail server.
+     * @param properties additional properties.
+     * @return a configured properties instance.
+     */
+    public static Properties get(String protocol, String host, String port, Map<String, String> properties)
+    {
+        return get(protocol, host, port, CONNECTION_TIMEOUT, READ_TIMEOUT, 0L, properties);
+    }
+
+    /**
+     * Configures a new properties instance with the specified parameters.
+     *
+     * @param protocol          the default protocol to be used.
+     * @param host              the host of the mail server.
+     * @param port              the port of the mail server.
+     * @param connectionTimeout the connection timeout.
+     * @param readTimeout       the read timeout.
+     * @param writeTimeout      the write timeout.
+     * @param properties        additional properties.
+     * @return a configured properties instance.
+     */
+    public static Properties get(String protocol, String host, String port, long connectionTimeout, long readTimeout, long writeTimeout, Map<String, String> properties)
```

Comment:
having this two overloads of the get method is a smell that this should actually be a builder. It's also not clear for me why to have this just like a "properties factory" instead of having it as a Session builder.